internal/server/server.go

```diff
@@ -58,7 +74,30 @@ func New(cfg config.Config) (*Server, error) {
 	}
 	s.authPool = authPool
 
-	s.poolMgr, err = NewPoolManager(s.cfg, s.logger)
+	if cfg.MetricsAddress != "" {
+		s.metrics = newMetrics()
+		router := mux.NewRouter()
+		s.metricsSrv = &http.Server{
+			Addr:         cfg.MetricsAddress,
+			Handler:      router,
+			ReadTimeout:  60 * time.Second,
+			WriteTimeout: 60 * time.Second,
+			IdleTimeout:  30 * time.Second,
+		}
+		router.HandleFunc("/health", s.healthHandler)
+		router.Handle("/metrics", s.metrics.metricsHandler())
+
+		s.metricsWg.Add(1)
+		go func() {
+			defer s.metricsWg.Done()
+			err2 := s.metricsSrv.ListenAndServe()
+			if err2 != nil && !errors.Is(err2, http.ErrServerClosed) {
+				s.logger.Printf("Error while running metrics server: %v\n", err2)
```

Comment:
I only have nits for you today
```suggestion
				s.logger.Printf("Error while running metrics server: %w\n", err2)
```

Reply:
As above, this is fine.